src/executables/locator/shared.ts

```diff
@@ -0,0 +1,36 @@
+import path = require('path');
+import * as fs from 'fs-extra';
+
+export function getVersionFromPath(rPath: string): string {
+    if (process.platform === 'win32') {
+        // not sure how to do this
+        return '';
+    } else {
+        try {
+            const scriptPath = path.normalize(`${rPath}/../Rcmd`);
+            const rCmdFile = fs.readFileSync(scriptPath, 'utf-8');
+            const regex = /(?<=R_VERSION=)[0-9.]*/g;
+            const version = regex.exec(rCmdFile)?.[0];
+            return version ?? '';
+        } catch (error) {
+            return '';
+        }
+    }
+}
+
+export function getArchitectureFromPath(path: string): string {
+    if (process.platform === 'win32') {
+        // \\bin\\i386 = 32bit
+        // \\bin\\x64 = 64bit
+        return '';
+    } else {
+        return '64-bit';
+    }
+}
+
```

Comment:
Would it make sense to use `R.version` (inside R) for this?

Reply:
That would definitely be more straightforward. The main reason I avoided that was because we would then need to spawn an R process for every single executable found

Reply:
I wonder what the load is like for Rscript? Alternatively, it looks like the R version is also exposed for windows by the `Rversion.h` header file in /R/include/